### PR TITLE
Remove PYTHONPATH from the simulator invocation

### PR DESCRIPTION
**Pull request by themperek**
- Simplify the makefile and remove `PYTHONPATH` from the simulator invocation
- Append "" to sys.path when python is embedded that is normally included.
- Resolve minor issue with` PYTHONPATH` order  `PYTHONPATH := $(PYTHONPATH):$(PWD)`
- It is on the user to append and export `PYTHONPATH` if needed

### CODE REVIEW

File: examples/adder/tests/Makefile [hunk 1]
```diff
@@ -37,7 +37,7 @@ else
 WPWD=$(shell pwd)
 endif
 
-PYTHONPATH := $(PWD)/../model:$(PYTHONPATH)
+export PYTHONPATH := $(PWD)/../model:$(PYTHONPATH)
```
Comment (eric-wieser) [inline examples/adder/tests/Makefile:40]: It's very tempting to remove this in a future patch and just use `import model.something` in the test.

Comment (themperek) [reply examples/adder/tests/Makefile:40]: Agree. Much nicer.  In the test, one can append to `sys.path` also rather than this. I can do in this PR?

Comment (eric-wieser) [reply examples/adder/tests/Makefile:40]: CI has already passed here, may as well merge this one first.

**Comment by eric-wieser** [comment]
Some typos in the commit message, will fix when I merge

**Comment by eric-wieser** [comment]
Hmm, on second thoughts this should have had a release note, since any user makefiles that set `PYTHONPATH` without `export` will now fail to work.

**Comment by themperek** [comment]
@eric-wieser Ths is causing an **error** on all simulator that start in `SIM_BUILD` folder `cd $(SIM_BUILD)`

It was discussed here: https://github.com/cocotb/cocotb/pull/1063 and somehow decided.

Should revert this or make a patch to align all simulators?
On the bad side, it will produce all empty files in test directory @imphil .

**Comment by eric-wieser** [comment]
It sounds like this means that `os.getcwd()` is set differently on different simulators, which is bad because it breaks python code attempting to `open("some_data_file")`.

Can you point to which simulators do that?

**Comment by themperek** [comment]
https://github.com/cocotb/cocotb/search?q=%22cd+%24%28SIM_BUILD%29%22&unscoped_q=%22cd+%24%28SIM_BUILD%29%22

We should rather align all of them. Since this is also issued for where `results.xml` is located. This is not related to this issues caused by this.

**Comment by eric-wieser** [comment]
So to summarize, we already fixed this issue in questa, but the other files are lagging behind?

That seems like a clear argument for changing those simulators to match.

**Comment by themperek** [comment]
Ok, will do.

**Pull request merged**